util-images/probes/pkg/dnsPropagation/metrics.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dnsPropagation
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/perf-tests/util-images/probes/pkg/common"
+)
+
+var (
+	// DNSPropagationSeconds calculate the propagation time of a given statefulSet the average difference between the pod running and pod discoverable timestamps.
+	DNSPropagationSeconds = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: common.ProbeNamespace,
+		Name:      "dns_propagation_seconds",
+		Help:      "Gauge of the time (in seconds) it took for the pods in a statefulSet to be discoverable after they start running.",
+	}, []string{"namespace", "service", "podName"})
+	// DNSPropagationCount calculate the number of DNS propagation checks performed.
+	DNSPropagationCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: common.ProbeNamespace,
+		Name:      "dns_propagation_count",
+		Help:      "Counter of the number of DNS propagation checks performed.",
+	}, []string{"namespace", "service", "podName"})
```

Comment:
should not be better to use a histogram https://prometheus.io/docs/concepts/metric_types/#histogram

Reply:
the reason here is that we don't really have bins per se, since we have exactly one measurement per label. Also timestamp of the measurement isn't important, so I wanted to choose a metric that wouldn't get it confused with histogram to store a single value per label.